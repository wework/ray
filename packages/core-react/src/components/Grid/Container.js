import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

export default function Container({ className, ...props }) {
  const classes = clsx('ray-page-container', className);

  return <div className={classes} {...props} />;
}

Container.propTypes = {
  className: PropTypes.string
};
