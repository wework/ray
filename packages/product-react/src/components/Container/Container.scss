$lucia-container-margin-1: 0 12px;
$lucia-container-padding-1: 0 40px;
$lucia-container-padding-2: 0 64px;
$lucia-container-fixed-margin-right: 52px;
$lucia-container-cell-width-fixed: 280px;

/* Overrides */

/* tablet */
@media (max-width: 1194px) and (min-width: 600px) {
  .ray-page-container {
    padding: $lucia-container-padding-1;
  }
}

/* > 1440px */
@media screen and (min-width: 1195px) {
  .ray-page-container {
    padding: $lucia-container-padding-1;
  }
}

/* > 1920 */
@media screen and (min-width: 1695px) {
  .ray-page-container {
    padding: $lucia-container-padding-2;
  }
}

/* Lucia Specific for Fixed Left Column */
.lucia-page-container-fixed-left {
  /* > 1440px */
  @media screen and (min-width: 1195px) {
    .ray-grid__cell:first-child,
    .ray-grid__cell:nth-child(10n) {
      margin-right: $lucia-container-fixed-margin-right;
      width: $lucia-container-cell-width-fixed;
    }

    .ray-grid__cell--span-1 {
      width: calc(12.5% - 67px);
    }
  }

  /* > 1680px */
  @media screen and (min-width: 1380px) {
    .ray-grid__cell:first-child {
      margin-right: $lucia-container-fixed-margin-right;
      width: $lucia-container-cell-width-fixed;
    }

    .ray-grid__cell--span-1,
    .ray-grid__cell:nth-child(10n) {
      margin: $lucia-container-margin-1;
      width: calc(8.3333% - 45px);
    }
  }

  /* > 1920px */
  @media (min-width: 1695px) {
    padding: $lucia-container-padding-2;

    .ray-grid__cell--span-1 {
      width: calc(9.2% - 57px);
    }
  }
}

/* Lucia Debug ( debug={true} ) */
.ray-page-container--debug {
  &.ray-page-container {
    background: yellow;
  }

  .ray-grid__cell {
    background: hotpink;
  }
}
