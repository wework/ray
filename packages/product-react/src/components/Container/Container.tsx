import clsx from 'clsx';
import React, { ReactHTML, ReactNode } from 'react';

export enum ContainerAlign {
  LEFT = 'left',
  RIGHT = 'right',
}

export enum ContainerJustify {
  LEFT = 'left',
  CENTER = 'center',
  RIGHT = 'right',
}

export type ContainerProps = {
  align?: ContainerAlign;
  as?: keyof ReactHTML;
  center?: any;
  children?: ReactNode;
  className?: string;
  debug?: boolean;
  fixed?: boolean;
  justify?: ContainerJustify;
} & typeof defaultProps;

const containerJustifyClass = (justify?: ContainerJustify) => {
  switch (justify) {
    case ContainerJustify.LEFT:
      return 'ray-page-container—justify-left';
    case ContainerJustify.CENTER:
      return 'ray-page-container—justify-center';
    case ContainerJustify.RIGHT:
      return 'ray-page-container—justify-right';
    default:
      return '';
  }
};

const containerAlignClass = (align?: ContainerAlign) => {
  switch (align) {
    case ContainerAlign.LEFT:
      return 'ray-page-container—align-left';
    case ContainerAlign.RIGHT:
      return 'ray-page-container—align-right';
    default:
      return '';
  }
};

export const Container = ({
  align,
  as: ComponentType,
  children,
  className,
  debug,
  fixed,
  justify,
  ...rest
}: ContainerProps) => (
  <ComponentType
    className={clsx(
      'ray-page-container',
      { 'ray-page-container--debug': debug },
      { 'lucia-page-container-fixed-left': fixed },
      containerJustifyClass(justify),
      containerAlignClass(align),
      className,
    )}
    {...rest}
  >
    {children}
  </ComponentType>
);

Container.ContainerAlign = ContainerAlign;
Container.ContainerJustify = ContainerJustify;

const defaultProps = {
  as: 'div',
};

Container.defaultProps = defaultProps;

