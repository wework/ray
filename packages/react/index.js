export { Container, Grid, Cell } from './src/components/Grid';
