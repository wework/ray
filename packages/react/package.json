{
  "name": "@wework/ray-react",
  "version": "0.0.1",
  "main": "index.js",
  "scripts": {
    "dev": "yarn storybook",
    "storybook": "start-storybook -p 6007 --ci"
  },
  "license": "Apache-2.0",
  "dependencies": {
    "@wework/ray-core": "1.3.1"
  },
  "devDependencies": {
    "@babel/core": "7.2.2",
    "@storybook/addon-knobs": "5.0.11",
    "@storybook/addon-options": "5.0.11",
    "@storybook/addon-viewport": "5.0.11",
    "@storybook/addons": "5.0.11",
    "@storybook/core": "5.0.11",
    "@storybook/react": "5.0.11",
    "babel-loader": "8.0.5",
    "clsx": "1.0.4",
    "prop-types": "15.7.2",
    "react": "16.8.6",
    "react-dom": "16.8.6",
    "storybook-chromatic": "1.3.3"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src/**/*"
  ],
  "keywords": [
    "wework",
    "react",
    "design-system"
  ]
}
