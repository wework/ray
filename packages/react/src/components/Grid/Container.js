import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

const Container = ({ children, className, debug, ...props }) => {
  const classes = clsx({
    'ray-page-container': true,
    'ray-page-container--debug': !!debug,
    [`${className}`]: !!className
  });

  return (
    <div className={classes} {...props}>
      {children}
    </div>
  );
};

Container.defaultProps = {
  debug: false
};

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  debug: PropTypes.bool
};

export default Container;
