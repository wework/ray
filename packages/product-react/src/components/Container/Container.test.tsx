import { mount, ReactWrapper } from 'enzyme';
import * as React from 'react';
import { Container } from './Container';

describe('Container', () => {
  let component: ReactWrapper<any>;

  beforeEach(() => {
    component = mount(<Container />);
  });

  test('should match snapshot and styles', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contain default class & passed in className', () => {
    component.setProps({ className: 'foo' });
    const container = component.find('div').at(0);
    expect(container.hasClass('ray-page-container')).toBe(true);
    expect(container.hasClass('foo')).toBe(true);
  });

  test('should accept "as" prop for setting element', () => {
    component.setProps({ as: 'main' });
    expect(component.find('div').length).toBe(0);
    expect(
      component
        .find('main')
        .at(0)
        .hasClass('ray-page-container'),
    ).toBe(true);
  });

  test('should pass additional props to component', () => {
    component.setProps({ id: 'foo', 'data-info': 'bar' });
    const container = component.find('.ray-page-container').at(0);
    expect(container.prop('id')).toEqual('foo');
    expect(container.prop('data-info')).toEqual('bar');
  });

  describe('Defaults', () => {
    test('should contain default class', () => {
      expect(
        component
          .find('div')
          .at(0)
          .hasClass('ray-page-container'),
      ).toBe(true);
    });
  });

  describe('Justifying', () => {
    test('should add class to justify left', () => {
      component.setProps({ justify: 'left' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—justify-left'),
      ).toBe(true);
    });
    test('should add class to justify center', () => {
      component.setProps({ justify: 'center' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—justify-center'),
      ).toBe(true);
    });
    test('should add class to justify right', () => {
      component.setProps({ justify: 'right' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—justify-right'),
      ).toBe(true);
    });
    test('should NOT add class to incorrect justify type', () => {
      component.setProps({ justify: 'notLeftCenterOrRight' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—justify-left'),
      ).toBe(false);
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—justify-center'),
      ).toBe(false);
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—justify-right'),
      ).toBe(false);
    });
  });

  describe('Alignment', () => {
    test('should add class to align left', () => {
      component.setProps({ align: 'left' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—align-left'),
      ).toBe(true);
    });
    test('should add class to align right', () => {
      component.setProps({ align: 'right' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—align-right'),
      ).toBe(true);
    });
    test('should NOT add class to incorrect align type', () => {
      component.setProps({ align: 'notLeftOrRight' });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—align-left'),
      ).toBe(false);
      expect(
        component
          .find('.ray-page-container')
          .hasClass('ray-page-container—align-right'),
      ).toBe(false);
    });
  });

  describe('Fixed Left', () => {
    it('should add correct class when "fix" prop is true', () => {
      expect(
        component
          .find('.ray-page-container')
          .hasClass('lucia-page-container-fixed-left'),
      ).toBe(false);
      component.setProps({ fixed: true });
      expect(
        component
          .find('.ray-page-container')
          .hasClass('lucia-page-container-fixed-left'),
      ).toBe(true);
    });
  });
});
