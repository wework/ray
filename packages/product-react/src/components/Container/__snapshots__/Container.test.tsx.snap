// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match snapshot and styles 1`] = `
<Container
  as="div"
>
  <div
    className="ray-page-container"
  />
</Container>
`;
